Fw/Buffer/Buffer.cpp

```diff
@@ -0,0 +1,177 @@
+// ======================================================================
+// \title  Buffer.cpp
+// \author mstarch
+// \brief  cpp file for Fw::Buffer implementation
+//
+// \copyright
+// Copyright 2009-2020, by the California Institute of Technology.
+// ALL RIGHTS RESERVED.  United States Government Sponsorship
+// acknowledged.
+//
+// ======================================================================
+#include <Fw/Buffer/Buffer.hpp>
+#include <Fw/Types/Assert.hpp>
+#include <Fw/Types/BasicTypes.hpp>
+
+#if FW_SERIALIZABLE_TO_STRING
+    #include <Fw/Types/EightyCharString.hpp>
+#endif
+#include <cstring>
+
+namespace Fw {
+
+Buffer::Buffer(): Serializable(),
+    m_serialize_repr(),
+    m_data(NULL),
+    m_size(0),
+    m_context(0xFFFFFFFF)
+{}
+
+Buffer::Buffer(const Buffer& src) : Serializable(),
+    m_serialize_repr(src.m_data, src.m_size),
+    m_data(src.m_data),
+    m_size(src.m_size),
+    m_context(src.m_context)
+{}
+
+Buffer::Buffer(U8* data, U32 size, U32 context) : Serializable(),
+    m_serialize_repr(data, size),
+    m_data(data),
+    m_size(size),
+    m_context(context)
+{}
+
+Buffer& Buffer::operator=(const Buffer& src) {
+    // Ward against self-assignment
+    if (this != &src) {
+        this->set(src.m_data, src.m_size, src.m_context);
+    }
+    return *this;
+}
+
+bool Buffer::operator==(const Buffer& src) const {
+    return (this->m_data == src.m_data) && (this->m_size == src.m_size) && (this->m_context == src.m_context);
+}
+
+U8* Buffer::getData() const {
+    return this->m_data;
+}
+
+U32 Buffer::getSize() const {
+    return this->m_size;
+}
+
+U32 Buffer::getContext() const {
+    return this->m_context;
+}
+
+void Buffer::setData(U8* const data) {
+    this->m_data = data;
+    if (m_data != NULL) {
+        this->m_serialize_repr.setExtBuffer(m_data, m_size);
+    }
+}
+
+void Buffer::setSize(const U32 size) {
+    this->m_size = size;
+    if (m_data != NULL) {
+        this->m_serialize_repr.setExtBuffer(m_data, m_size);
+    }
+}
+
+void Buffer::setContext(const U32 context) {
+    this->m_context = context;
+}
+
+void Buffer::set(U8* const data, const U32 size, const U32 context) {
+    this->m_data = data;
+    this->m_size = size;
+    if (m_data != NULL) {
+        this->m_serialize_repr.setExtBuffer(m_data, m_size);
+    }
+    this->m_context = context;
+}
+
+Fw::SerializeBufferBase& Buffer::getSerializeRepr() {
```

Comment:
This is the one I think we should change. I would not have the `resetSer()` call and just return the buffer. The user can reset it if that's what they want to do. They could have logic that appends to what is there, and this would break it.

Reply:
Good catch.  That was my intention, it should be fixed now.